include/Operator/SecondQuantized/USQHamiltonian.hpp

```diff
@@ -0,0 +1,250 @@
+    }
+
+    /*
+     *  PUBLIC METHODS
+     */
+
+    /**
+     *  @return the dimension of the Hamiltonian, i.e. the number of spinors in which it is expressed
+     */
+    size_t dimension() const { return this->sq_hamiltonian_alpha.dimension(); }
+
+    /**
+     *  @return the total of the alpha contributions of the unrestricted Hamiltonian
+     */
+    const SQHamiltonian<Scalar>& alphaHamiltonian() const { return this->sq_hamiltonian_alpha; }
+
+    /**
+     *  @return the total of the beta contributions of the unrestricted Hamiltonian
+     */
+    const SQHamiltonian<Scalar>& betaHamiltonian() const { return this->sq_hamiltonian_beta; }
+
+    /**
+     *  @return the total contributions to the mixed alpha & beta two-electron part of the unrestricted Hamiltonian
+     */
+    const ScalarSQTwoElectronOperator<Scalar>& twoElectronMixed() const { return this->total_two_op_mixed; }
+
+    /**
+     *  @return the total contributions to the mixed alpha & beta two-electron part of the unrestricted Hamiltonian
+     */
+    const std::vector<ScalarSQTwoElectronOperator<Scalar>>& twoElectronContributionsMixed() const { return this->two_op_mixed; }
+
+    /**
+     *  In-place transform the matrix representations of Hamiltonian
+     *
+     *  @param T    the transformation matrix between the old and the new orbital basis
+     */
+    void transform(const TransformationMatrix<Scalar>& T) {
+
+        this->sq_hamiltonian_alpha.transform(T);
+        this->sq_hamiltonian_beta.transform(T);
+
+        // Transform the mixed
+        this->total_two_op_mixed.transform(T);
+        for (auto& two_op : this->two_op_mixed) {
+            two_op.transform(T);
+        }
+    }
+
+    /**
+     *  In-place transform the matrix representations of the alpha components of the unrestricted Hamiltonian
+     *
+     *  @param T    the transformation matrix between the old and the new orbital basis
+     */
+    void transformAlpha(const TransformationMatrix<Scalar>& T) {
+
+        this->sq_hamiltonian_alpha.transform(T);
+        // Transform the mixed
+        auto new_two_electron_parameters = this->total_two_op_mixed.parameters();
+        // transform the two electron parameters "g_aabb" to "g_a'a'bb"
+        new_two_electron_parameters.template matrixContraction<Scalar>(T, 0);
+        new_two_electron_parameters.template matrixContraction<Scalar>(T, 1);
+        this->total_two_op_mixed = ScalarSQTwoElectronOperator<Scalar> ({new_two_electron_parameters});
+
+        for (auto& two_op : this->two_op_mixed) {
+            auto new_two_electron_parameters = two_op.parameters();
+            // transform the two electron parameters "g_aabb" to "g_a'a'bb"
+            new_two_electron_parameters.template matrixContraction<Scalar>(T, 0);
+            new_two_electron_parameters.template matrixContraction<Scalar>(T, 1);
+            two_op = ScalarSQTwoElectronOperator<Scalar> ({new_two_electron_parameters});
```

Comment:
I think you also noticed that this code fragment contains duplicate code. Wouldn't it be possible to transfer the implementation of such a 'transformation' into `SQTwoElectronOperator`? 

I'm just brainstorming: it might even be beneficial to implement a `USQTwoElectronOperator` in this PR. It would then inherit from `SQTwoElectronOperator`, but its sole responsibility is to implement these `mixedTransform()` methods.

Reply:
For an issue